packages/application-to-register-webapp/src/routes/__tests__/land/legal-agreement-type.spec.js

```diff
@@ -0,0 +1,132 @@
+import constants from '../../../utils/constants'
+const url = '/land/legal-agreement-type'
+
+describe(url, () => {
+  const redisMap = new Map()
+
+  describe('GET', () => {
+    it(`should render the ${url.substring(1)} view without any selection`, async () => {
+      const response = await submitGetRequest({ url })
+      expect(response.statusCode).toBe(200)
+    })
+
+    it(`should render the ${url.substring(1)} view with conservation selected`, async () => {
+      jest.isolateModules(async () => {
+        redisMap.set(constants.redisKeys.LEGAL_AGREEMENT_DOCUMENT_TYPE, 'Conservation covenant')
+        let viewResult, contextResult
+        const legalAgreementDetails = require('../../land/legal-agreement-type')
+        const request = {
+          yar: redisMap,
+          info: {
+            referrer: 'check-legal-agreement-details'
+          }
+        }
+        const h = {
+          view: (view, context) => {
+            viewResult = view
+            contextResult = context
+          }
+        }
+        await legalAgreementDetails.default[0].handler(request, h)
+        expect(viewResult).toEqual('land/legal-agreement-type')
+        expect(contextResult.conservationType).toEqual(true)
+      })
+    })
+
+    it(`should render the ${url.substring(1)} view with planning selected`, async () => {
+      jest.isolateModules(async () => {
+        redisMap.set(constants.redisKeys.LEGAL_AGREEMENT_DOCUMENT_TYPE, 'Planning obligation (section 106 agreement)')
+        let viewResult, contextResult
+        const legalAgreementDetails = require('../../land/legal-agreement-type')
+        const request = {
+          yar: redisMap,
+          info: {
+            referrer: 'agreement-details'
+          }
+        }
+        const h = {
+          view: (view, context) => {
+            viewResult = view
+            contextResult = context
+          }
+        }
+        await legalAgreementDetails.default[0].handler(request, h)
+        expect(viewResult).toEqual('land/legal-agreement-type')
+        expect(contextResult.planningObligationType).toEqual(true)
+      })
+    })
+
+    it(`should render the ${url.substring(1)} view with dont have selected`, async () => {
+      jest.isolateModules(async () => {
+        redisMap.set(constants.redisKeys.LEGAL_AGREEMENT_DOCUMENT_TYPE, 'I do not have a legal agreement')
+        let viewResult, contextResult
+        const legalAgreementDetails = require('../../land/legal-agreement-type')
+        const request = {
+          yar: redisMap,
+          info: {
+            referrer: 'check-legal-agreement-details'
+          }
+        }
+        const h = {
+          view: (view, context) => {
+            viewResult = view
+            contextResult = context
+          }
+        }
+        await legalAgreementDetails.default[0].handler(request, h)
+        expect(viewResult).toEqual('land/legal-agreement-type')
+        expect(contextResult.dontHave).toEqual(true)
```

Comment:
Is this test temporary until the kickout functionality reflects the prototype?

If somebody has indicated that they do not have a legal agreement the prototype redirects to a page that says they need one. As such I believe you should not be able to reach the legal agreement page using a cached value of 'I do not have a legal agreement'.